presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxTableHandle.java

```diff
@@ -36,21 +36,21 @@
     private final List<String> objectNames;
     private final List<JmxColumnHandle> columnHandles;
     private final boolean liveData;
-    private final List<HostAddress> addresses;
+    private final List<Node> nodes;
 
     @JsonCreator
     public JmxTableHandle(
             @JsonProperty("tableName") SchemaTableName tableName,
             @JsonProperty("objectNames") List<String> objectNames,
             @JsonProperty("columnHandles") List<JmxColumnHandle> columnHandles,
             @JsonProperty("liveData") boolean liveData,
-            @JsonProperty("addresses") List<HostAddress> addresses)
+            @JsonProperty("nodes") List<Node> nodes)
```

Comment:
Why hold on to the list of nodes in the TableHandle? I'd be better to just hold on to the predicate and filter them out during split generation.

Reply:
In my understanding, to hold the predicate ( `TupleDomain` or `Map<ColumnHandle, Domain>`) in the TableHandle, we need to make all classes held in the `ValueSet` serializable to JSON. Is this correct?

Reply:
Oops, I didn't know `TupleDomain` is always JSON serializable. I fixed to hold `TupleDomain` in `TableHandle` instead of the list of node.